### PR TITLE
Huaxi/fluent folder

### PR DESCRIPTION
**Pull request by yumikohey**
#### Pull request checklist

- [ X ] Addresses an existing issue: Fixes #0000
- [ X ] Include a change request file using `$ npm run change`

#### Description of changes
Provide props to override folder colors;
Provide props to swap metadata and shared icon's position within folder cover.
screen shot 2019-02-13 at 10 07 23 am

#### Focus areas to test
Cover Folder.
(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/7976)

### CODE REVIEW

File: packages/experiments/src/components/FolderCover/FolderCover.scss [hunk 1]
```diff
@@ -72,14 +72,14 @@
...
+  @include margin-left(8px);
   flex: 1 1;
-  @include text-align(right);
+  @include text-align(left);
   white-space: nowrap;
   text-overflow: ellipsis;
   overflow: hidden;
   font-weight: $ms-font-weight-regular;
-  color: $ms-color-white;
+  color: $ms-color-yellowDark;
```
Comment (ThomasMichon) [inline packages/experiments/src/components/FolderCover/FolderCover.scss]: The catch is we need to switch between 'fluent' and 'non-fluent' flavors via prop, or else an app that isn't fluent-ready will get a mismatch in styling.

File: packages/experiments/src/components/FolderCover/initializeFolderCovers.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { registerIcons, IIconOptions } from '@uifabric/styling';
 
-const ASSET_CDN_BASE_URL = 'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/foldericons';
+const ASSET_CDN_BASE_URL = 'https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/foldericons-fluent';
```
Comment (ThomasMichon) [inline packages/experiments/src/components/FolderCover/initializeFolderCovers.tsx]: Host apps may already override the base URL to a different subfolder using `initializeFolderCovers()`.

File: packages/experiments/src/components/signals/Signal.scss [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
   .dark &,
   .dark.selected {
-    color: $ms-color-white;
+    color: $ms-color-yellowDark;
```
Comment (ThomasMichon) [inline packages/experiments/src/components/signals/Signal.scss]: Look at the `SignalsPage` and switch to 'Is Dark'. Not sure we want them all to be yellow!
"Dark mode" for Signals makes them render white against a dark background for a media tile.

File: packages/experiments/src/components/FolderCover/FolderCover.tsx [hunk 4]
```diff
@@ -73,11 +73,26 @@ export class FolderCover extends React.Component<IFolderCoverProps, IFolderCover
       metadata,
       signal,
       children,
+      isFluent,
       ...divProps
     } = this.props;
 
     const assets = ASSETS[size][type];
+    const MetadataIcon = (): JSX.Element | null => {
```
Comment (ThomasMichon) [inline packages/experiments/src/components/FolderCover/FolderCover.tsx]: React trivia:
When you define an inline function, and then create a React element around it, it messes up React's diff logic. Reason being that React only compares the 'type', and it's a new function every time!
For this behavior, I suggest not creating functions at all. Instead, just do:
```ts
const metadataIcom = metadata ? (
  <span className={ ... }>
    { metadata }
  </span>
) : null;
...
```
Then just emit as:
```ts
<React.Fragment>
  { metadataIcon }
  { signalIcon }
</React.Fragment>
```

Comment (kenotron) [reply packages/experiments/src/components/FolderCover/FolderCover.tsx]: Agreed with Thomas there. Also, you're creating new component here, why promote them as real functional components?

File: packages/experiments/src/components/FolderCover/initializeFolderCovers.tsx (hunk 2 above)
Comment (kenotron) [reply packages/experiments/src/components/FolderCover/initializeFolderCovers.tsx]: Does look like this is resolved with a revert.

Comment (yumikohey) [reply packages/experiments/src/components/FolderCover/initializeFolderCovers.tsx]: good catch! updated!

**Pull request merged**